File: CafeEase/UI/cafeease_desktop/lib/models/order_item.dart
import 'package:json_annotation/json_annotation.dart';

part 'order_item.g.dart';

@JsonSerializable()
class OrderItem {
  int? id;
  int? orderId;
  int? productId;
  int? quantity;
  double? price;
  String? productName;
  String? size;
  String? milkType;
  int? sugarLevel;
  String? note;


  OrderItem({
    this.id,
    this.orderId,
    this.productId,
    this.quantity,
    this.price,
    this.productName,
    this.size,
    this.milkType,
    this.sugarLevel,
    this.note
  });

  factory OrderItem.fromJson(Map<String, dynamic> json) =>
      _$OrderItemFromJson(json);

  Map<String, dynamic> toJson() => _$OrderItemToJson(this);
}

File: CafeEase/CafeEase.Model/OrderItem.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace CafeEase.Model
{
    public class OrderItem
    {
        public int Id { get; set; }
        public int OrderId { get; set; }
        public int ProductId { get; set; }
        public string ProductName { get; set; }
        public int Quantity { get; set; }
        public decimal Price { get; set; }
        public string? Size { get; set; }
        public string? MilkType { get; set; }
        public int? SugarLevel { get; set; }
        public string? Note { get; set; }
    }
}
